pkg/core/blockchain.go

```diff
@@ -2313,6 +2321,16 @@ func (bc *Blockchain) SubscribeForExecutions(ch chan *state.AppExecResult) {
 	bc.subCh <- ch
 }
 
+// SubscribeForPersistedBlockHeaders adds given channel to new header event broadcasting, so
+// when there is a new header added to the chain you'll receive it via this
+// channel. Make sure it's read from regularly as not reading these events might
+// affect other Blockchain functions. Make sure you're not changing the received
+// headers, as it may affect the functionality of Blockchain and other
+// subscribers.
+func (bc *Blockchain) SubscribeForPersistedBlockHeaders(ch chan *block.Header) {
```

Comment:
OK, we likely need a consistency between RPC subscription ID and this API name. Let's rename it to `SubscribeForHeadersOfAddedBlocks` and `UnsubscribeFromHeadersOfAddedBlocks` to be more precise.